docs/decisions/0005-toggle-life-expectancy.rst

```diff
@@ -0,0 +1,93 @@
+
+The second point was proposed to force people to be more intentional around "permanent" toggles, so someone could not simply mark a toggle as "permanent" to avoid the clean-up required for "temporary" toggles without hopefully being a bit more intentional.
+
+In the `PR to implement this proposal`_, the solution was tweaked based on some lost decisions. The update adjusted the ``toggle_uses_cases`` to the following values:
+
+* *temporary*: A new use case that replaced all uses case options that were considered temporary.
+* *circuit_breaker*: Replaced "graceful_degradation", although the OEP has not yet been updated.
+* *vip, opt_out, opt_in, open_edx*: The other original permanent use case options.
+
+The current implementation has the following flaws:
+
+* Defining "permanent" as the absence of "temporary" reduces clarity for "permanent" toggles.
+* The usage of use cases isn't clear.
+
+  * The options "circuit_breaker" and "opt_out" ("opt_in") seem like two types of toggles. The former is ops related, and the latter is business related.
+
+    * Note that "opt_out" and "opt_in" only differ in language based on the default of the toggle. For example, if the default is True, then "opt_out" linguistically makes sense as the purpose of the toggle, and if the default is False, "opt_in" makes more sense.
+
+  * The options "vip" vs "open_edx" seem more about the scope at which one might apply the ability to opt-out/opt-in. However, this scope may vary depending on the Open edX instance, and it may not make sense to document the scope selected for edX.org. Additionally, the ``toggle_implementation`` annotation may duplicate this information, assuming the scoping features of the implementation match the intent.
+  * In practice, it seems that "open_edx" is often chosen as a general option for permanent toggles, and negates the intent to use ``toggle_use_cases`` to force intentional design and consideration for permanent toggles.
+
+.. _set of use cases for toggles: https://github.com/edx/open-edx-proposals/blob/c2d3b2a/oeps/oep-0017-bp-feature-toggles.rst#use-cases
+.. _Details a reporting requirement: https://github.com/edx/open-edx-proposals/blob/c2d3b2a/oeps/oep-0017-bp-feature-toggles.rst#req-12-report
+.. _PR discussion about toggle_use_cases: https://github.com/edx/edx-platform/pull/24815#issuecomment-681174018
+.. _PR to implement this proposal: https://github.com/edx/code-annotations/pull/49
+
+Decision
+--------
+
+In order to make the toggle life expectancy more clear, and the justification
+for permanent toggles more clear, we will take the following actions:
+
+
+* As originally proposed, introduce ``toggle_life_expectancy`` with values of "temporary" or "permanent". This makes the options very clear.
+* Replace ``toggle_use_cases`` with ``toggle_permanent_justification``.
+
+  * ``toggle_permanent_justification`` would be required for "permanent" toggles.
```

Comment:
From @nasthagiri 
> Clarifying question: is toggle_permanent_justification a free-form value?  Or would there be pre-defined [set of choices]?

Reply:
Thanks for the questions.

Using code-annotations, by default we have a choice between:
1. free-form, or 
2. a strict list of choices.
With additional work, we could add a custom linting rule on top of the free-form entry.

However, I believe free-form will be good enough.

For both these stakeholders, I think first priority would be getting rid of "temporary" toggles.

However, when the time comes to review "permanent" toggles to see if they really should be permanent, here is what would happen with today's annotations:

* A stakeholder would see that `.. toggle_use_cases: openedx` (the most popular option today for permanent toggles).
* There would be no way to easily tell how confident we are that this really should be a permanent toggle.
* Either we assume it really is permanent, or we do lots of research or use DEPR to ensure it is really should be permanent.
* If we determine it really should be permanent, again, there is no consistent way to note the new confidence level.

With the proposed change, here is what might happen:

* A stakeholder might see `toggle_permanent_justification` with any of the following values:
  * See ADR detailing why this toggle must be permanent: ...
  * See OSPR that added this feature and explains why it is needed for different Open edX instances: ...
  * Tried DEPR-XXX, but got push back because ... 
  * None
  * Seems permanent, but toggle defined before justification was required.
  * ...
* For each of these, the stakeholder could decide whether the justification is strong enough, or whether to initiate DEPR to test the strength of the justification.
* Stakeholders could also add a prefix like "[Director Approved]", or something else that hopefully won't get copy/pasted, so that the same toggles don't need to be reviewed multiple times once determined to have a strong justification.

I plan to add some additional color to the ADR based on this comment.

Reply:
@timmc-edx: When you get a chance to review, I added a commit to enhance the ADR based on this response. See https://github.com/edx/edx-toggles/pull/129/commits/e280b63e123f1bd13e2cccbeb3b618d89f9a1e89